projects/update_success_nightly_validation.kp/knowledge.md

```diff
@@ -0,0 +1,383 @@
+    Previous buildId mismatch: 20170906220306 must be older than 20170905220108:	0.002%
+    Previous buildId mismatch: 20170906220306 must be older than 20170906100107:	0.001%
+    Previous buildId mismatch: 20170907100318 must be older than 20170903100443:	0.001%
+    Previous buildId mismatch: 20170907100318 must be older than 20170905100117:	0.001%
+    Previous buildId mismatch: 20170907100318 must be older than 20170905220108:	0.002%
+    Previous buildId mismatch: 20170907100318 must be older than 20170906100107:	0.002%
+    Previous buildId mismatch: 20170907100318 must be older than 20170906220306:	0.003%
+    Previous buildId mismatch: 20170907194642 must be older than 20170903220032:	0.001%
+    Previous buildId mismatch: 20170907194642 must be older than 20170905100117:	0.001%
+    Previous buildId mismatch: 20170907194642 must be older than 20170906100107:	0.001%
+    Previous buildId mismatch: 20170907194642 must be older than 20170906220306:	0.001%
+    Previous buildId mismatch: 20170907194642 must be older than 20170907100318:	0.001%
+    Previous buildId mismatch: 20170907220212 must be older than 20170903220032:	0.001%
+    Previous buildId mismatch: 20170907220212 must be older than 20170905100117:	0.001%
+    Previous buildId mismatch: 20170907220212 must be older than 20170905220108:	0.001%
+    Previous buildId mismatch: 20170907220212 must be older than 20170906100107:	0.001%
+    Previous buildId mismatch: 20170907220212 must be older than 20170907100318:	0.002%
+    Previous buildId mismatch: 20170908100218 must be older than 20170907100318:	0.001%
+    Previous buildId mismatch: 20170908100218 must be older than 20170907220212:	0.001%
+    Previous buildId mismatch: 20170908220146 must be older than 20170907100318:	0.001%
+    Previous buildId mismatch: 20170908220146 must be older than 20170907220212:	0.001%
+    Previous buildId mismatch: 20170908220146 must be older than 20170908100218:	0.001%
+    Previous channel mismatch: expected nightly-cck-google got nightly:	0.001%
+    Previous channel mismatch: expected nightly-cck-mint got nightly:	0.003%
+    Previous channel mismatch: expected nightly-cck-mozillaonline got nightly:	0.001%
+    Previous channel mismatch: expected nightly-cck-yandex got nightly:	0.001%
+
+
+The vast majority of the data in the payload seems reasonable (99.87%).
+
+However, a handful of `update` pings are reporting a `previousBuildId` with `None`: this is unexpected. **After discussing this with the *update team*, it seems like this could either be due to Nigthly channel weirdness or to the customization applied by the [CCK tool](https://mike.kaply.com/cck2/).** Additionally, some pings are reporting a `previousChannel` different than the one in the environment: this is definitely due to the CCK tool, given the *cck* entry in the channel name. These issues do not represent a problem, as most of the data is correct and their volume is fairly low.
```

Comment:
The data doesn't match this conclusion any more. Now there are no update pings with `previousBuildId == None`.

Reply:
Yeah, sorry about that. The explanation still stands though, so I've only changed the comment a bit.